### PR TITLE
743 infer legend prop

### PR DESCRIPTION
**Pull request by Jesus89**
Fixes https://github.com/CartoDB/cartoframes/issues/743

### CODE REVIEW

File: cartoframes/viz/legend.py [hunk 1]
```diff
@@ -42,35 +42,52 @@ def _init_legend(self, data):
...
             if isinstance(_type, dict) and geom_type in _type:
                 _type = _type.get(geom_type)
+            self._check_type(_type)
+
+            if _type and not _prop:
+                _prop = self._infer_prop(_type)
+
+            self._check_prop(_prop)
+
+            print(_prop)
```
Comment (elenatorro) [inline cartoframes/viz/legend.py]: Remove `print`

File: cartoframes/viz/legend.py [hunk 2]
```diff
@@ -42,35 +42,52 @@ def _init_legend(self, data):
...
+
+    def _check_type(self, _type):
+        if _type and _type not in constants.LEGEND_TYPES:
+            raise ValueError(
+                'Legend type is not valid. Valid legend types are: {}.'.format(
+                    ', '.join(constants.LEGEND_TYPES)
+                ))
+
+    def _check_prop(self, _prop):
+        print(_prop)
```
Comment (elenatorro) [inline cartoframes/viz/legend.py]: Remove `print`

Comment (elenatorro) [reply cartoframes/viz/legend.py]: As an improvement, I'd move `_type` and `_prop` initialization to separated methods and I'd leave the responsibility of checking if they're correct to these methods:

```python
def _get_type(self, geom_type):
    _type = self._type.get(geom_type)
        if isinstance(self._type, dict) and geom_type in self._type
        else self._type
  
    self._check_type(self._type)
    return _type
```

```python
def _get_prop(self, _type):
    _prop = self._prop if self._prop else self._infer_prop(_type)
    
    self._check_prop(_prop)
    return _prop
```

And then, in `get_info`:

```python
def get_info(self, geom_type=None):
   # ...
    _type = self._get_type(geom_type)
    _prop = self._get_prop(_type)
```

**Comment by Jesus89** [comment]
Thanks! Suggestions added.

Note: I have updated the changelog of this feature in #741

**Pull request merged**